src/main/java/nwoolcan/controller/brewery/BreweryControllerImpl.java

```diff
@@ -20,8 +37,71 @@ public BreweryControllerImpl(final Brewery brewery) {
         this.warehouseController = new WarehouseControllerImpl(brewery.getWarehouse());
     }
 
+    @Override
+    public Collection<MasterBatchViewModel> getBatches(final QueryBatch queryBatch) {
+        return brewery.getBatches(queryBatch)
+                      .stream()
+                      .map(MasterBatchViewModel::new)
+                      .collect(Collectors.toList());
+    }
+
+    @Override
+    public Result<Empty> stockBatch(final int batchId, final int beerArticleId, final Date expirationDate) {
+        final Result<Batch> batchResult = getBatchById(batchId);
+        final Result<BeerArticle> beerArticleResult = getBeerArticleById(beerArticleId);
+        return Result.ofEmpty()
+                     .flatMap(() -> batchResult)
+                     .flatMap(() -> beerArticleResult)
+                     .flatMap(() -> brewery.stockBatch(batchResult.getValue(),
+                                                       beerArticleResult.getValue(),
+                                                       expirationDate));
+    }
+
+    @Override
+    public Result<Empty> stockBatch(final int batchId, final int beerArticleId) {
+        final Result<Batch> batchResult = getBatchById(batchId);
+        final Result<BeerArticle> beerArticleResult = getBeerArticleById(beerArticleId);
+        return Result.ofEmpty()
```

Comment:
I (and @Carlovan) think that's not clear. You are using flatmaps like a bunch of ifs. If you don't like the other solution I think it's clearer to use plain if `if(beerArticleResult.isPresent() && batchResult.isPresent()) ...`. Those flatmaps are not mapping anything

Reply:
If/else are "Not gut" cit Oleg. I use require instead. Give it a look.

Reply:
don't argue about "self preferences". 
we have little time and anyway these are little implementation things that we don't need to argue about.
everyone is responsible for its own code, don't try to impose self preferences and reject pr for these things